ast/compile.go

```diff
@@ -4376,6 +4370,16 @@ func rewriteVarsInRef(vars ...map[Var]Var) func(Ref) Ref {
 	}
 }
 
-func rewriteVarsNop(node Ref) Ref {
-	return node
+// NOTE(sr): contrary to rewriteVarsInRef, this helper creates
+// a copy of the Ref, and replaces the vars in there.
+func replaceRewritten(r Ref, rw map[Var]Var) Ref {
+	r0 := r.Copy()
+	for i, r := range r {
+		if v, ok := r.Value.(Var); ok {
```

Comment:
Only thought here is to use a transformer because if we have something like `r[[__local0__]]` then it won't be handled as is.